### PR TITLE
Add support for users

### PR DESCRIPTION
**Pull request by mwasilew2**
## Description of the change

This is part of "User Management" set of features. It adds to the terraform provider support for managing user mappings. See the relevant clickup issue for more details.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

> Fix [#1]()

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [x] Examples for new resources and data sources have been added
- [x] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [x] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Comment by mwasilew2** [comment]
This is blocked on API changes necessary on the backend side

**Comment by mwasilew2** [comment]
blocked by: https://github.com/spacelift-io/backend/pull/5658

**Comment by mwasilew2** [comment]
The backend PR was merged, the tests are passsing, this is ready for review

File: spacelift/provider.go [hunk 1]
```diff
@@ -125,7 +126,7 @@ func Provider(commit, version string) plugin.ProviderFunc {
 				"spacelift_stack_aws_role":               resourceStackAWSRole(),           // deprecated
 				"spacelift_stack_gcp_service_account":    resourceStackGCPServiceAccount(), // deprecated
 				"spacelift_terraform_provider":           resourceTerraformProvider(),
-				"spacelift_idp_group_mapping":            resourceIdpGroupMapping(),
+				"spacelift_user_mapping":                 resourceUserMapping(),
```
Comment (michieldewilde) [inline spacelift/provider.go:129]: it should be called `spacelift_user` this is the name we are using in the UI.

File: spacelift/resource_user_mapping_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+					Attribute("invitation_email", Equals(exampleEmailNew)),
+					Attribute("username", Equals(newUsername)),
+					SetContains("policy", "root"),
+					SetContains("policy", "ADMIN"),
+				),
+			},
+		})
+	})
+
+	t.Run("can remove one access", func(t *testing.T) {
```
Comment (michieldewilde) [inline spacelift/resource_user_mapping_test.go:63]: incomplete test?

**Review by michieldewilde — CHANGES_REQUESTED** [review]
Only the naming is an issue.

File: spacelift/resource_user_mapping_test.go (hunk 2 above)
Comment (mwasilew2) [reply spacelift/resource_user_mapping_test.go:63]: oopsie xD

**Comment by mwasilew2** [comment]
ping @michieldewilde

**Review by michieldewilde — APPROVED** [review]

**Pull request merged**